File: C++/825. Friends Of Appropriate Ages.cpp
class Solution {
public:
    int numFriendRequests(vector<int>& a) {
        map<int,int>m;
        int c=0;
        for(int i:a)
        m[i]++;
        for(auto i=m.begin();i!=m.end();i++){
            for(auto j=m.begin();j!=m.end();j++){
                if((i->first*0.5+7)>=j->first || i->first<j->first ||(i->first<100 && 100<j->first))
                continue;
                c+=i->second*j->second;
                if(i->first==j->first)
                c-=i->second;
            }
        }
        return c;
    }
};

File: JAVA/825. Friends Of Appropriate Ages.java
class Solution {
    public int numFriendRequests(int[] a) {
        Map<Integer, Integer> m = new HashMap<>();
        int c = 0;
        for (int age : a) {
            m.put(age, m.getOrDefault(age, 0) + 1);
        }
        for (Map.Entry<Integer, Integer> i : m.entrySet()) {
            for (Map.Entry<Integer, Integer> j : m.entrySet()) {
                if ((i.getKey() * 0.5 + 7) >= j.getKey() || i.getKey() < j.getKey() || (i.getKey() < 100 && 100 < j.getKey())) {
                    continue;
                }
                c += i.getValue() * j.getValue();
                if (i.getKey().equals(j.getKey())) {
                    c -= i.getValue();
                }
            }
        }
        return c;
    }
}